tests/test_integration.py

```diff
@@ -89,10 +91,48 @@ def test_good_result_manifests():
     """
     result = run_query("test_all_result_groups", Namespace())
 
-    for group, count in result["data"]:
+    for group, _ in result["data"]:
         if group is None:
             continue
 
         assert (
             not task.is_bad_group("x", group) and "\\" not in group
         ), f"{group} group is bad!"
+
+
+def test_caching_of_tasks(tmp_path):
+    config_file = tmp_path / "config.toml"
```

Comment:
Nit: you could make the adr config a fixture so other tests in the future can make use of it

Reply:
I believe I handled this.